arrow-array/src/array/union_array.rs

```diff
@@ -226,9 +226,10 @@ impl UnionArray {
     ///
     /// # Panics
     ///
-    /// Panics if the `type_id` provided is less than zero or greater than the number of types
+    /// Panics if the `type_id` provided is less than zero or greater than or equal the number of types
```

Comment:
```suggestion
    /// Panics if the `type_id` provided is not present in the array's DataType
```
As the type signature can skip over type ids